internal/metric/async.go

```diff
@@ -41,11 +41,9 @@ type AsyncCollector interface {
 type AsyncInstrumentState struct {
 	lock sync.Mutex
 
-	// errorHandler will be called in case of an invalid
-	// metric.AsyncRunner, i.e., one that does not implement
-	// either the single-or batch-runner interfaces.
-	errorHandler func(error)
-	errorOnce    sync.Once
+	// errorOnce will use the global.Handler to report an error once of an
+	// invalid runner being attempted to run.
+	errorOnce sync.Once
```

Comment:
```suggestion
	// errorOnce will use the global.Handler to report an error
	// once in case of an invalid runner attempting to run.
	errorOnce sync.Once
```